openstudiocore/src/energyplus/ForwardTranslator/ForwardTranslateCoilHeatingDesuperheater.cpp

```diff
@@ -104,13 +101,40 @@ boost::optional<IdfObject> ForwardTranslator::translateCoilHeatingDesuperheater(
   boost::optional<ModelObject> heatingSource = modelObject.heatingSource();
 
   if ( heatingSource ) {
-      boost::optional<IdfObject> _heatingSource = translateAndMapModelObject(heatingSource.get());
 
-      if ( _heatingSource && _heatingSource->name() )
-      {
-        object.setString(Coil_Heating_DesuperheaterFields::HeatingSourceObjectType,_heatingSource->iddObject().name());
-        object.setString(Coil_Heating_DesuperheaterFields::HeatingSourceName,_heatingSource->name().get());
+    boost::optional<IdfObject> _heatingSource = translateAndMapModelObject(*heatingSource);
+
+    if (_heatingSource) {
+
+      // Note JM 2019-03-14:
+      // If the coil in question is a DX coil (CoilCoolingDXSingleSpeed, CoilCoolingDXTwoSpeed, CoilCoolingDXTwoStageWithHumidityControlMode)
+      // and this DX coil isn't already wrapped in a Unitary, then the FT will wrap it into a CoilSystem:Cooling:DX object and return that, but we
+      // need the DX coil here and not the wrapper.
+      //
+      // Note: Other accepted types are Refrigeration objects and don't suffer the same problem
+      // (Refrigeration:Condenser:AirCooled, Refrigeration:Condenser:EvaporativeCooled, Refrigeration:Condenser:WaterCooled)
+
+      std::string objectType;
+      std::string objectName;
+
+      if (_heatingSource->iddObject().type() == IddObjectType::CoilSystem_Cooling_DX) {
```

Comment:
@jmarrec  I don't know this syntax nor what this translator is trying to accomplish, just a question here. What if the desuperheater coil is in a UnitarySystem?

Reply:
The coil is translated and returned, and it's the nodes of the coil (and not the parent unitary) that end up there. Does that make sense?

Source:
* https://github.com/NREL/OpenStudio/blob/b32c3d6fe31f7401a0be860ed301039b99ea98d9/openstudiocore/src/energyplus/ForwardTranslator.cpp#L973
* [translateCoilDXSingleSpeedWithoutUnitary](https://github.com/NREL/OpenStudio/blob/b32c3d6fe31f7401a0be860ed301039b99ea98d9/openstudiocore/src/energyplus/ForwardTranslator/ForwardTranslateCoilCoolingDXSingleSpeed.cpp#L53) you'll see it's the coil that's returned. The naming of the method can be confusing because it's this one that's called when the coil is already inside a unitary, and if not then another method is called where a CoilSystemDX is added ("With unitary")